src/main/java/edu/ie3/util/StringUtils.java

```diff
@@ -102,4 +114,42 @@ public static String quote(String input) {
   public static String cleanString(String input) {
     return input.replaceAll("[^\\w]", "_");
   }
+
+  /**
+   * Quotes a given string that contains special characters to comply with the csv specification RFC
+   * 4180 (https://tools.ietf.org/html/rfc4180). Double quotes in JSON strings are escaped with the
+   * same character to make the csv data readable later.
+   *
+   * @param inputString string that should be converted to a valid rfc 4180 string
+   * @param csvSep separator of the csv file
+   * @return a csv string that is valid according to rfc 4180
+   */
+  public static String csvString(String inputString, String csvSep) {
+    if (needsCsvRFC4180Quote(inputString, csvSep)) {
+      /* clean the string by first quoting start and end of the string and then replace all double quotes
+       * that are followed by one or more double quotes with single double quotes */
+      String quotedStartEndString = quote(inputString).replaceAll("\"\"*", "\"");
+      /* get everything in between the start and end quotes and replace single quotes with double quotes */
+      String stringWOStartEndQuotes =
+          quotedStartEndString
+              .substring(1, quotedStartEndString.length() - 1)
+              .replaceAll("\"", "\"\"");
+      /* finally add quotes to the strings start and end again */
+      return quote(stringWOStartEndQuotes);
```

Comment:
This implementation seems not to go the straight way from my point of view. I would suggest to do the following:

```suggestion
      /* Get rid of first and last quotation if there is some. */
      String inputUnquoted = unquote(inputString);
      /* Escape every double quotation mark within the String by doubling it */
      String withEscapedQuotes = inputUnquoted.replaceAll("\"", "\"\"");
      /* finally add quotes to the strings start and end again */
      return quote(withEscapedQuotes);
```

whereby a new `unquote(String)` method has to be implemented:
```
public static String unquote(String input) {
  return input.matches("^\".*\"$") ? substring(1, input.length() - 1) : input;
}
```